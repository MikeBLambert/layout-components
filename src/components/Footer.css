.icons {
  margin-top: 25vh;
  display: flex;
  flex-direction: row;
  justify-content: space-around;

  i {
    flex-basis: auto;
    font-size: 70px;
  }
  #gmail:hover {
    color: red;
    }
  #linkedin:hover {
    color: blue;
  }
  #github{
    opacity: 0.9;
    &:hover {
      opacity: 1;
    }
  }
}