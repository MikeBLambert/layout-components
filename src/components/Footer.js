import Component from './Component';
import styles from './Footer.css';

export default class Footer extends Component {

  render() {
    const contactIcons = this.props.contactIcons.map(icon => {
      return `<i class="${icon}"></i>`;
    });

    return `
    <footer class="${styles.icons}">${contactIcons.join(' ')}</footer>
    `;
  }
}
