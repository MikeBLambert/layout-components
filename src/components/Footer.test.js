import Footer from './Footer';
import styles from './Footer.css';

describe('Footer Component', () => {

  it('should be a footer', () => {
    const footer = new Footer({
      ccontactIcons: [
        { class: 'far fa-envelope', id: 'gmail' },
        { class:'fab fa-linkedin-in', id: 'linkedin' },
        { class: 'fab fa-github', id: 'github' }]
    });
    expect(footer.render()).toMatchSnapshot();
  });
})