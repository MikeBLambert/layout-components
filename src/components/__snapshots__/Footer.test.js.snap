// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should be a footer 1`] = `
"
    <footer>
      <h1>Contact</h1>
      <div class=\\"icons\\">
        
      <a id=\\"undefined\\">
        <i class=\\"undefined\\"></i>
      </a>
       
      <a id=\\"undefined\\">
        <i class=\\"undefined\\"></i>
      </a>
       
      <a id=\\"undefined\\">
        <i class=\\"undefined\\"></i>
      </a>
      
      </div>
    </footer>
    "
`;
